Documentation/network/standalone.rst

```diff
@@ -0,0 +1,1493 @@
+the path taken from the fabric to the L4 load-balancer to the subsequent backends.
+
+For the PCAP recorder, a wildcarded n-tuple filtering for IPv4 and IPv6 has been
+implemented and exposed as a Hubble API. This allows for building wildcard-filter
+rules with:
+
+- Arbitrary source address prefix
+- Arbitrary destination address prefix
+- Individual source port number or all ports
+- Individual destination port number or all ports
+- Individual protocols (TCP/UDP) or all protocols
+
+From the installed rules, masks are derived and filter rules are inserted into
+a hash table. While there may be a small set of individual masks, there can be
+millions of filter entries in the hash table which fit the constraint of
+the masks. The set of masks is small and limited to a maximum of 32, as the
+more masks are necessary, the slower the fast-path becomes given classifying and
+subsequently capturing traffic might incur processing overhead.
+
+The datapath iterates through the masks to create a temporary tuple and performs
+a subsequent hash table lookup to find a matching entry. The filters match for the
+inbound packet. If there has been a match, then the outbound packet is marked to
+be pushed to the recorder as well.
+
+The PCAP recorder works with both DSR as well as SNAT operation modes.
+
+In order to enable the PCAP recorder in the agent, it must be built into
+the datapath (``--enable-recorder=true``).
+
+Hubble can be used to access the recorder and expose a recorder API under
+``unix:///var/run/cilium/hubble.sock`` for local clients to connect to:
+
+.. parsed-literal::
+
+   docker run --name cilium-gateway -td \\
+     -v /sys/fs/bpf:/sys/fs/bpf \\
+     -v /lib/modules:/lib/modules \\
+     --privileged=true \\
+     --network=host \\
+     quay.io/cilium/cilium:|IMAGE_TAG| \\
+     cilium-agent \\
+     --enable-ipv4=true \\
+     --enable-ipv6=true \\
+     --datapath-mode=lb-only \\
+     --bpf-lb-acceleration=native \\
+     --enable-recorder=true \\
+     --enable-hubble=true \\
+     --enable-hubble-recorder-api=true \\
+     --devices=bond0
+
+Hubble's `recorder command-line interface <https://github.com/cilium/hubble/blob/master/cmd/record/record.go>`_
+communicates to the agent through this exposed API. For simplicity, we use the
+command-line tool in this example, but third party orchestration tooling can
+use the API by connecting to the unix domain socket directly in order to build
+a distributed PCAP recorder infrastructure among a group of load-balancers.
+
+For the latter, it is recommended to expose Cilium's runtime directory to the host
+via Docker volumes (``-v /var/run/cilium/:/var/run/cilium/``), so that other control
+plane software can talk to ``unix:///var/run/cilium/hubble.sock``.
+
+By default, Hubble stores recorded PCAPs for post-analysis under ``/var/run/cilium/pcaps/``.
+
+Example for recording all TCP-based traffic for the node for a time of one second:
```

Comment:
One comment here is that hubble won't actually dump _all_ the traffic inside pcap: if a `cilium_capture_in` isn't matched by `cilium_capture_out`. One case is when traffic is dropped by the host. The cilium event will be sent in any case by `cilium_capture_in`, so a custom control plane can dump such traffic as well

Reply:
makes sense, added comment